internal/manifest/manifest.go

```diff
@@ -140,87 +142,139 @@ func validateHexString(value HexString, expectedNumBytes int) error {
-		return fmt.Errorf("TdAttributes is invalid: %w", err)
+		errs = append(errs, fmt.Errorf("TdAttributes is invalid: %w", err))
 	}
 	if err := validateHexString(r.Xfam, 8); err != nil {
-		return fmt.Errorf("Xfam is invalid: %w", err)
+		errs = append(errs, fmt.Errorf("Xfam is invalid: %w", err))
 	}
 	for i, rtmr := range r.Rtrms {
 		if err := validateHexString(rtmr, 48); err != nil {
-			return fmt.Errorf("Rtmr[%d] is invalid: %w", i, err)
+			errs = append(errs, fmt.Errorf("Rtmr[%d] is invalid: %w", i, err))
 		}
 	}
-	return nil
+	return errors.Join(errs...)
 }
 
 // Validate checks the validity of all fields in the manifest.
 func (m *Manifest) Validate() error {
+	var errs []error
 	for policyHash, policy := range m.Policies {
 		if err := policy.Validate(policyHash); err != nil {
-			return fmt.Errorf("validating policy %s: %w", policyHash, err)
+			errs = append(errs, fmt.Errorf("validating policy %s: %w", policyHash, err))
 		}
 	}
 
 	if err := m.ReferenceValues.Validate(); err != nil {
-		return fmt.Errorf("validating reference values: %w", err)
+		errs = append(errs, newValidationError("validating reference values", err))
 	}
 
 	for _, keyDigest := range m.WorkloadOwnerKeyDigests {
 		if _, err := keyDigest.Bytes(); err != nil {
-			return fmt.Errorf("decoding key digest %s: %w", keyDigest, err)
+			errs = append(errs, fmt.Errorf("decoding key digest %s: %w", keyDigest, err))
 		} else if len(keyDigest) != hex.EncodedLen(sha256.Size) {
-			return fmt.Errorf("workload owner key digest %s has invalid length: %d (expected %d)", keyDigest, len(keyDigest), hex.EncodedLen(sha256.Size))
+			errs = append(errs, fmt.Errorf("workload owner key digest %s has invalid length: %d (expected %d)", keyDigest, len(keyDigest), hex.EncodedLen(sha256.Size)))
 		}
 	}
 
 	for _, key := range m.SeedshareOwnerPubKeys {
 		if _, err := ParseSeedShareOwnerKey(key); err != nil {
-			return fmt.Errorf("invalid seed share owner public key %s: %w", key, err)
+			errs = append(errs, fmt.Errorf("invalid seed share owner public key %s: %w", key, err))
 		}
 	}
-	return nil
+	return errors.Join(errs...)
+}
+
+// validationError wraps an error with a prefix message.
+//
+// The error will be printed indented below the message.
+// Errors joined by errors.Join will be printed on separate lines with the same indentation.
+type validationError struct {
+	msg string
+	err error
+}
+
+func newValidationError(msg string, err error) *validationError {
+	return &validationError{msg: msg, err: err}
+}
+
+func (e *validationError) Error() string {
+	return e.formatError(0)
+}
+
+func (e *validationError) formatError(indent int) string {
+	prefix := strings.Repeat("  ", indent)
+	var sb strings.Builder
+	sb.WriteString(prefix + e.msg + ":")
+
+	if e.err == nil {
+		return sb.String()
+	}
+
+	var errs []error
+	// Interface is implemented by errors returned by errors.Join.
+	if unwrapErr, ok := e.err.(interface{ Unwrap() []error }); ok {
```

Comment:
There's also `Unwrap() error` which we'd need to check, see https://pkg.go.dev/errors#pkg-overview.